packages/jest-console/src/matchers.js

```diff
@@ -8,58 +8,66 @@ import { matcherHint, printExpected, printReceived } from 'jest-matcher-utils';
  */
 import supportedMatchers from './supported-matchers';
 
+const createErrorMessage = ( spyInfo ) => {
+	const { spy, pass, calls, matcherName, methodName, expected } = spyInfo;
+	const hint = pass ? `.not${ matcherName }` : matcherName;
+	const message = pass
+		? `Expected mock function not to be called but it was called with:\n${ calls.map(
+				printReceived
+		  ) }`
+		: `Expected mock function to be called${
+				expected ? ` with:\n${ printExpected( expected ) }\n` : '.'
+		  }\nbut it was called with:\n${ calls.map( printReceived ) }`;
+
+	return () =>
+		`${ matcherHint( hint, spy.getMockName() ) }` +
+		'\n\n' +
+		message +
+		'\n\n' +
+		`console.${ methodName }() should not be used unless explicitly expected\n` +
```

Comment:
Nit: can we add a full stop at the end of the sentence? :)
```suggestion
		`console.${ methodName }() should not be used unless explicitly expected.\n` +
```

This is how it looks at the moment.
<img width="1076" alt="Screenshot 2023-08-30 at 11 16 36" src="https://github.com/WordPress/gutenberg/assets/699132/6fbe5f00-00ac-445a-b2ef-722e66abaf98">